src/kafunk/ProduceBuffer.fs

```diff
@@ -0,0 +1,49 @@
+﻿namespace Kafunk
+
+open FSharp.Control
+open Kafunk
+
+type BufferingProducer  = private {
+
+  ///The producer
+  producer : Producer
+
+  ///The buffer
+  buffer : Buffer<ProducerMessage>
+  
+  consume : ProducerMessage -> bool }
+
+and BufferType = 
+    | Blocking
+    | Discarding
+
+[<Compile(Module)>]
+module BufferingProducer = 
+
+  let private Log = Log.create "Kafunk.BufferingProducer"
+  
+  ///Create a producer with buffer
+  let createBufferingProducer (producer:Producer) (buffertype:BufferType) (capacity:int) (batchSize:int) (batchTimeMs:int) (timeIntervalMs:int) (errorHandle:(ProducerMessage seq -> unit) option)= 
+
+    let consume (producer: Producer) (batch: ProducerMessage seq): Async<unit> = async {
+      try
+        do! Producer.produceBatched producer batch |> Async.Ignore
```

Comment:
Rather than discarding the ProducerResult we could publish that as an Event on the producer buffer. Thoughts?

Reply:
How about the produceresult of this result event, sounds like a dead loop